### PR TITLE
New-DbaAvailabilityGroup - Add config for FailureConditionLevel and change the default

### LINKED ISSUE #6605: New-DbaAvailabilityGroup change parameter FailureConditionLevel default value
### Summary of new feature

Currently the parameter FailureConditionLevel of New-DbaAvailabilityGroup defaults to 1 (OnServerDown). When deploying an Availability Group you might get unexpected results testing Availability Groups. Especially when comparing a deployment that followed the Microsoft documentation (Wizard or T-SQL). 
I would like to see the default FailureConditionLevel using the same default for this parameter as CREATE AVAILABILITY GROUP (T-SQL) - https://docs.microsoft.com/en-us/sql/t-sql/statements/create-availability-group-transact-sql?view=sql-server-ver15 - Failure Condition Level 3: 
>  "Specifies that an automatic failover should be initiated on critical SQL Server internal errors, such as orphaned spinlocks, serious write-access violations, or too much dumping. _This is the default behavior._". 

### Proposed technical details (if applicable)

Instead of the default that is currently applied `[string]$FailureConditionLevel = "OnServerDown"`, this should be changed to `[string]$FailureConditionLevel = "OnCriticalServerErrors"` (Failure Condition Level 3).

### Latest version of dbatools as of writing

I currently use Version 1.0.113.

### PR DESCRIPTION
**Pull request by andreasjordan**
## Type of Change

 - [ ] Bug fix (non-breaking change, fixes # )
 - [ ] New feature (non-breaking change, adds functionality, fixes # )
 - [x] Breaking change (effects multiple commands or functionality, fixes #6605 )
 - [ ] Ran manual Pester test and has passed (`.\tests\manual.pester.ps1`)
 - [ ] Adding code coverage to existing functionality
 - [ ] Pester test is included
 - [ ] If new file reference added for test, has is been added to github.com/sqlcollaborative/appveyor-lab ?
 - [ ] Unit test is included
 - [ ] Documentation
 - [ ] Build system

Changed the default to the same CREATE AVAILABILITY GROUP uses, but now use a config to be able to change this default if needed.

### CODE REVIEW

File: functions/New-DbaAvailabilityGroup.ps1 [hunk 1]
```diff
@@ -246,7 +256,7 @@ function New-DbaAvailabilityGroup {
         [ValidateSet('None', 'Primary', 'Secondary', 'SecondaryOnly')]
         [string]$AutomatedBackupPreference = 'Secondary',
         [ValidateSet('OnAnyQualifiedFailureCondition', 'OnCriticalServerErrors', 'OnModerateServerErrors', 'OnServerDown', 'OnServerUnresponsive')]
-        [string]$FailureConditionLevel = "OnServerDown",
+        [string]$FailureConditionLevel = (Get-DbatoolsConfigValue -FullName 'AvailabilityGroups.Default.FailureConditionLevel' -Fallback 'OnCriticalServerErrors'),
```
Comment (wsmelton) [inline functions/New-DbaAvailabilityGroup.ps1:259]: I sure hope release notes have a big caveat as this being a breaking change.

**Comment by andreasjordan** [comment]
We have another parameter with differrent default:

$agClusterType = 'Wsfc'                                           # Default in CREATE AVAILABILITY GROUP (as far as I see it)
#$agClusterType = 'External'                                      # Default in New-DbaAvailabilityGroup

Would like to add a config for that as well and change the default.

**Comment by andreasjordan** [comment]
And another one:

#$replicaConnectionModeInSecondaryRolePS = 'AllowAllConnections'  # Default in New-DbaAvailabilityGroup
$replicaConnectionModeInSecondaryRolePS = 'AllowNoConnections'    # Default in CREATE AVAILABILITY GROUP

File: functions/New-DbaAvailabilityGroup.ps1 (hunk 1 above)
Comment (andreasjordan) [reply functions/New-DbaAvailabilityGroup.ps1:280]: I also updated the DESCRIPTION

**Comment by potatoqualitee** [comment]
Thank you so much 🙇🏼

**Pull request merged**